content/zh/blog/2023/native-sidecars/index.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Istio 中的 Kubernetes Native Sidecar
+description: 使用 Istio 演示新的 SidecarContainers 功能。
+publishdate: 2023-08-15
+attribution: "John Howard (Google); Translated by Wilson Wu (DaoCloud)"
+keywords: [istio,sidecars,kubernetes]
+---
+
+如果您曾经听说过有关服务网格的些许信息，
+那应该就是它们使用 Sidecar 模式进行工作：代理服务器与您的应用程序代码部署在一起。
+Sidecar 就是这样：它是一种模式。到目前为止，在 Kubernetes 中还没有对 Sidecar 容器进行正式支持。
+
+这导致了许多问题：如果您有一个按照预期终止的 Job，
+但由于 Sidecar 容器却不能达到这个预期，该怎么办？这是
+[Kubernetes Issue 跟踪器上最受欢迎的](https://github.com/kubernetes/kubernetes/issues/25908)具体使用场景。
+
+在 Kubernetes 中添加 Sidecar 支持的提案于 2019 年正式提出。
+一路走来，经历了许多停顿，在去年该项目重新启动后，对 Sidecar
+的正式支持将以 Alpha 功能发布在 Kubernetes 1.28 中。
+Istio 已经实现了对此功能的支持，在本文中您可以了解如何利用其优势。
+
+## Sidecar 的困境 {#sidecar-woes}
+
+Sidecar 容器提供了强大的功能，但也存在一些问题。
+虽然 Pod 内的容器可以共享一些东西，但它们的*生命周期*是完全解耦的。
+对于 Kubernetes 来说，这两个容器在功能上是相同的。
+
+然而，在 Istio 中，它们并不相同 — Istio 容器必须以主应用程序容器正常运行为前提，
+如果没有主应用程序容器，那么 Istio 容器就没有任何价值。
+
+这种期望的不匹配会导致各种问题：
+
+* 如果应用程序容器启动速度比 Istio 容器快，则无法访问网络。
+  这在 Istio 的 GitHub 上以压倒性优势赢得了[最多 +1](https://github.com/istio/istio/issues/11130)。
+* 如果 Istio 容器先于应用程序容器关闭，则应用程序容器无法访问网络。
+* 如果应用程序容器有意退出（通常是在 `Job` 中使用），Istio 容器仍将运行并保持 Pod 无限期运行。
+  这也是一个 [GitHub 中的顶级 Issue](https://github.com/istio/istio/issues/11659)。
+* 在 Istio 容器启动之前运行的 `InitContainers` 将无法访问网络。
+
+虽然 Istio 社区内外花费了无数时间来解决这些问题，但取得的成果仍然有限。
+
+## 从根源解决 {#fixing-the-root-cause}
+
+虽然 Istio 中日益复杂的变通方案可以帮助减轻 Istio 用户的痛苦，
+但理想情况下，所有相关情况都应该正常运转 - 而不仅仅是针对 Istio。
+幸运的是，Kubernetes 社区一直努力直接在 Kubernetes 中解决这些问题。
+
+在 Kubernetes 1.28 中，合并了一项为 Sidecar 添加原生支持的新功能，
+这为 5 年多的持续工作画上了句号。通过该功能，我们的所有问题都可以得到解决，而无需其他变通方案！
+
+当我们进入“GitHub Issue 名人堂”时，
```

Comment:
```suggestion
当我们进入 “GitHub Issue 名人堂”时，
```

Reply:
I think the full-width double quotation mark connect Chinese and English characters does not need space, is that right?